site/content/docs/contributing/project-structure.md

```diff
@@ -28,23 +59,123 @@ the root command [kind.go][kind.go] where we register other commands to build
+```
+// +k8s:deepcopy-gen=package
+// +k8s:conversion-gen=sigs.k8s.io/kind/pkg/cluster/config
+// +k8s:defaulter-gen=TypeMeta
+```
+
+Additionally, [pkg/cluster/config/types.go](pkg/cluster/config/types.go) has an additional tag:
+```
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+```
+
+Let’sseehow this tags work with the
+[kubernetes/code-generator](https://github.com/kubernetes/code-generator)
+binaries deepcopy-gen, defaulter-gen, and conversion-gen.
+
+For each of the directories related to defining a configuration for kind, we start by running [deepcopy-gen](https://godoc.org/k8s.io/code-generator/cmd/deepcopy-gen). deep-copy-gen generates functions that efficiently perform a full deep-copy of each type that is part of the configuration. These functions will be used to pass values to each type, go from a versioned configuration to the top level one.
+
+Once we have these utility functions in place then we will need to run
+[defaulter-gen](https://godoc.org/k8s.io/code-generator/cmd/defaulter-gen)
+to generate efficient defaulters (functions that will fill in default value for configuration fields) for the configuration schema based on the
+[`Config`](/pkg/cluster/config/types.go)
+and the [`Node`](/pkg/cluster/config/types.go) types.
+The way this works is that the
+`// +k8s:defaulter-gen=TypeMeta`
+comment tag will generate a defaulter for the `Config` type as this possesses a `TypeMeta` field.
+
+The `TypeMeta` field is defined in
+[k8s.io/apimachinery/pkg/apis/meta/v1](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#TypeMeta).
+`TypeMeta` is a struct with a Kind and APIVersion fields. Structures that are versioned or persisted should inline TypeMeta.
+
+
+The final sep in code generation for kind’s configuration specification involves running
+[conversion-gen](https://godoc.org/k8s.io/code-generator/cmd/conversion-gen)
+which will scan its input directories, looking at the package defined in each of those directories for comment tags that define a conversion code generation task. In Kind, you will see the following comment tag
+```
+// +k8s:conversion-gen=sigs.k8s.io/kind/pkg/cluster/config
+```
+which introduces a conversion task for which the destination package (the top level configuration definition in [kind/pkg/cluster/config](/pkg/cluster/config)).
+This last step build on the deep copy generators and defaulters previously created to enable kind to understand any configuration version.
+
+
+### Verify
+Now that you understand the update steps, the verify tasks should make more sense. You can check the compliance of the entire project by running the `verify-all.sh` script. This script will do the following:
+* check spelling using [client9/misspell](https://github.com/client9/misspell)
```

Comment:
Black Friday (the markdown parser used in hugo) needs a blank like before lists usually (and does in this case)

Reply:
Fixed